scipy/stats/stats.py

```diff
@@ -944,8 +944,17 @@ def moment(a, moment=1, axis=0):
             # the input was 1D, so return a scalar instead of a rank-0 array
             return np.float64(0.0)
     else:
-        mn = np.expand_dims(np.mean(a, axis), axis)
-        s = np.power((a - mn), moment)
+        # Subtract mean along the axis, compute element-wise square
+        a_zero_mean = a - np.expand_dims(np.mean(a, axis), axis)
+        a_zero_mean_2 = a_zero_mean**2
+        
+        s = a_zero_mean_2.copy()
+        for k in range(1, mom // 2):
```

Comment:
Apparently I did not copy the whole thing:

======================================================================
FAIL: test_skewtest (test_mstats_basic.TestCompareWithStats)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/stefan/ghproj/scipy/build/testenv/lib/python3.4/site-packages/scipy/stats/tests/test_mstats_basic.py", line 991, in test_skewtest
    assert_allclose(r[0], rm[0], rtol=1e-15)
  File "/usr/lib/python3/dist-packages/numpy/testing/utils.py", line 1183, in assert_allclose
    verbose=verbose, header=header)
  File "/usr/lib/python3/dist-packages/numpy/testing/utils.py", line 644, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-15, atol=0

(mismatch 100.0%)
 x: array(0.38677410656235006)
 y: array(0.3867741065623524)

======================================================================
FAIL: test_skewtest_2D_WithMask (test_mstats_basic.TestCompareWithStats)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/stefan/ghproj/scipy/build/testenv/lib/python3.4/site-packages/scipy/stats/tests/test_mstats_basic.py", line 1013, in test_skewtest_2D_WithMask
    assert_equal(r[0][0],rm[0][0])
  File "/usr/lib/python3/dist-packages/numpy/ma/testutils.py", line 100, in assert_equal
    raise AssertionError(msg)
AssertionError: 
Items are not equal:
 ACTUAL: 0.38677410656235006
 DESIRED: 0.38677410656235239

----------------------------------------------------------------------